### DISCUSSION TITLE
Badge

### DISCUSSION THREAD

**Discussion by flatcapco** [saadeghi/daisyui#3277] [Ideas / feature requests]
Hi the default badge class implements a fixed height which doesn't work so well when using with a longer badge text.
using h-auto and text-center will prevent the following issue:

current:
https://capture.dropbox.com/5aXaloiKAujIX6O3

suggested:
https://capture.dropbox.com/dz4y6bxuLs3PksKL

**Comment by saadeghi** [comment]
I see but none of the screenshots you posted look good.  
If the text doesn't fit, maybe you should give it more space to avoid text break? 🤔

**Reply by flatcapco** [reply]
I didnt say it looked good - it was a more sensible default, and whilst it can be optimised in a lot of situations its not always possible so this seems more appropriate than the current way?

**Reply by saadeghi** [reply]
I understand. There's a reason components are not stretching.  
Let's say you have multiple badges next to each other.  
They have different amount of text. Some have small icons some have bigger icons, some are only text.  
If there's no height, these badges would look awful. Each would have different heights just to fit the content...

The goal of a design system or a UI library never is to fit the content at all costs. The goal is to provide a system on how the components should look like and if the content doesn't fit in the design, the solution is not stretching the components. Instead we should consider changing the structure, spacing, or positioning.

This rule applies to all component libraries and all design systems.

Imagine the bottom navigation bar in iOS or Android. There's space for 3 to 6 icons usually. And each icon can have 1 or 2 words as label.  
What happens if I try to fit 12 icons with a paragraph of text for each label? It's not possible. Should the design system or UI library allow it or stretch the items? No.  
The solution is putting the content where there's enough space.  

Let me know if you have any questions.

**Reply by flatcapco** [reply]
I mean I do appreciate that, and of course theres lots of different contexts this could become a issue. 

Would you be opposed to using inline instead of inline-flex does that satisfy a better default? Its still ugly (no disagreement here) but it seems a more graceful or recognisable destruction of the badge as a worse case scenario?

https://capture.dropbox.com/9t7XdHNya81JuqvJ

Obviously I can handle this all myself I'm just trying to help suggest a more robust default

**Reply by saadeghi** [reply]
Can you show me the the whole block you're designing? I'm sure there's a better way to avoid this issue.

**Reply by flatcapco** [reply]
The data is driven from the database - a (user submitted tag cloud) for locations. 
Adding a tighter word count wouldn't work for the system and overflow (is messier in my opinion) so I need to allow certain badges to break over multiple lines (worse case scenario but possible since we don't control the order or number of these tags)

**Reply by saadeghi** [reply]
For a tag could if you're not limiting the characters, I would suggest something like this:  
https://play.tailwindcss.com/HTuVJibZCc

**Reply by flatcapco** [reply]
Thanks I'll try that - seems like the best solution
